test/new-e2e/tests/containers/base_test.go

```diff
@@ -157,7 +157,7 @@ func (suite *baseSuite) testMetric(args *testMetricArgs) {
 
 			metrics, err := suite.Fakeintake.FilterMetrics(
 				args.Filter.Name,
-				fakeintake.WithTags[*aggregator.MetricSeries](args.Filter.Tags),
+				fakeintake.WithMatchingTags[*aggregator.MetricSeries](args.Filter.Tags),
```

Comment:
Should we do the same change inside [`testLog` line 285](https://github.com/DataDog/datadog-agent/pull/23609/commits/38030051e50d65cb90eb917023c4dcb35376b308#diff-093798e4d908dc036e29010c541e499735184872af7e061cc64ef9bc1874a70aR285) ?

Reply:
I think we should leave it for another PR

Reply:
Let’s not forget to do it.
Because `testMetric` and `testLog` have extremely close API. So, having a different behavior would be extremely confusing.